src/ng/q.js

```diff
@@ -354,8 +354,8 @@ function qFactory(nextTick, exceptionHandler) {
       var that = this;
       var done = false;
       try {
-        if ((isObject(val) || isFunction(val))) then = val && val.then;
-        if (isFunction(then)) {
+        if(isPromiseLike(val)) {
+          then = val && val.then
```

Comment:
Why store a reference to `then` from `val` only to do `then.call(val, ...)` a couple lines below?  Seems like you should just be able to do `val.then(...args)` and avoid the `.call` requirement here.  It doesn't look like `then` is used anywhere but there.